src/python/pants/core/target_types.py

```diff
@@ -307,6 +308,34 @@ class GenericTarget(Target):
     )
 
 
+# -----------------------------------------------------------------------------------------------
+# `_target_generator_sources_helper` target
+# -----------------------------------------------------------------------------------------------
+
+
+class TargetGeneratorSourcesHelperSourcesField(MultipleSourcesField):
+    uses_source_roots = False
+    required = True
+
+
+class TargetGeneratorSourcesHelperTarget(Target):
```

Comment:
I get a bit hung up on the class name here.. not sure I can come up with anything better though, I'll take a stab at it, like: `TargetGeneratorDependencyHelperTarget`. It surely isn't any shorter, but it makes the sources field slightly more readable by avoiding repeating `Sources` in the name `TargetGeneratorDependencyHelperSourcesField`. 

I think that name better reflects what it is it helps with. Maybe.

Reply:
Agreed the name isn't great. 

I'm not super committed to the user facing name `_target_generator_sources_helper` either, but it was the best I could come up with. `_target_generator_helper` felt too vague?

I only used `Sources` in the class name to mirror the symbol name, which is the convention we normally follow.

Reply:
I don't think it's about "dependency_helper" though, it's really about tracking the source files for `--changed-since`. Sources is more improtant than dependencies here.